packages/debugger/src/service.ts

```diff
@@ -329,6 +329,36 @@ export class DebuggerService implements IDebugger, IDisposable {
     }
   }
 
+  /**
+   * Request to set a variable in the global scope.
+   *
+   * @param name The name of the variable.
+   * @param value The value of the variable.
+   */
+  async setVariableInGlobal(name: string, value: string): Promise<void> {
+    if (!this.session) {
+      throw new Error('No active debugger session');
+    }
+    const frames = this.model.callstack.frames;
+    this.session
+      .sendRequest('setExpression', {
+        expression: name,
+        value: value,
```

Comment:
Candid question but is this answering the need?

I tried with a complex object and it failed:

![image](https://user-images.githubusercontent.com/8435071/203773673-061d3be3-f69c-4a9e-aabe-f72e7401af21.png)

Note: I have no idea if it is possible to copy complex object (especially in a language independent way).

Reply:
Candid answer: maybe not :smile: 
I haven't tried it with a complex object, but in this implementation the value is sent in a request, so it will probably never work with complex objects. I'll see in the `debugpy` API if there is something similar on the kernel side.

Reply:
I haven't found anything like that in the DAP specifications.
I understand that the only way would be to evaluate some code, which would not be generic for languages.
This PR can be closed because it doesn't actually meet the need.
Sorry for your time @fcollonval

Reply:
One way to think about it is that we are already making addition to DAP (e.g. `inspectVariables` see https://jupyter-client.readthedocs.io/en/latest/messaging.html#additions-to-the-dap), but copying variables is a generic operation which ideally would be supported in the main DAP.
It looks that DAP only supports special evaluation with aim of copying to clipboard (`clipboard` context); this PR could be reworked to support that use case instead (which would still be an improvement for some scenarios, but not as good as the ability to copy to global namespace).

Can you open an issue over at https://github.com/microsoft/debug-adapter-protocol to see if DAP would be happy to add a "copy to another frame" request and link back from here so we can track it?

Reply:
Thanks for suggestion @krassowski. I opened an issue [here](https://github.com/microsoft/debug-adapter-protocol/issues/360)

Reply:
I converted the PR to draft as it will now wait for https://github.com/ipython/ipykernel/pull/1055 to be merged and released (an probably also https://github.com/jupyter-xeus/xeus-python/pull/562), which both wait for the next release of `debugpy`, which should integrate https://github.com/microsoft/debugpy/pull/1147.

Reply:
> It looks that DAP only supports special evaluation with aim of copying to clipboard (`clipboard` context); this PR could be reworked to support that use case instead (which would still be an improvement for some scenarios, but not as good as the ability to copy to global namespace).

Thanks, I'll probably open an other PR for this feature.